core/lib/prover_interface/src/inputs.rs

```diff
@@ -132,16 +135,50 @@ impl PrepareBasicCircuitsJob {
     }
 }
 
-/// Enriched `PrepareBasicCircuitsJob`. All the other fields are taken from the `l1_batches` table.
-#[derive(Debug, Clone)]
-pub struct BasicCircuitWitnessGeneratorInput {
-    pub block_number: L1BatchNumber,
-    pub previous_block_hash: H256,
-    pub previous_block_timestamp: u64,
-    pub block_timestamp: u64,
-    pub used_bytecodes_hashes: Vec<U256>,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VMRunWitnessInputData {
+    pub l1_batch_number: L1BatchNumber,
+    pub previous_root_hash: H256,
+    pub previous_meta_hash: H256,
+    pub previous_aux_hash: H256,
+    pub used_bytecodes: HashMap<U256, Vec<[u8; 32]>>,
     pub initial_heap_content: Vec<(usize, U256)>,
-    pub merkle_paths_input: PrepareBasicCircuitsJob,
+    pub protocol_version: ProtocolVersionId,
+    pub bootloader_code: Vec<[u8; 32]>,
+    pub default_account_code_hash: U256,
```

Comment:
That's just because of different methods all across the codebase which are returning/using different types. In our particular case these types are exactly ones that are accepted by BWG without any conversions

Reply:
But I agree that it's pretty nasty